File: espn/play_success.py
import pandas as pd
import numpy as np

pbp_folder = 'E:\\college football\\pbp'

pbp = pd.read_csv(pbp_folder + '\\csv\\' + 'espn_pbp.csv', header = 0)

for index, row in pbp.iterrows():
    if () & () & ():
        if row.down == 1:
            succ = 0.9718*np.exp(-0.031*row.dist)
        elif row.down == 2:
            succ = 0.9389*np.exp(-0.055*row.dist)
        elif row.down == 3:
            succ = 0.6256*np.exp(-0.071*row.dist)

File: espn/play_success.R
library(lme4)

pbp = read.csv('E:\\college football\\pbp\\csv\\espn_pbp.csv')

down1 = pbp[(pbp$down == 1) & (pbp$playtype %in% c('pass','rush')) & (pbp$yrd2end > 20),]
down1$succ_est = 0.9718*exp(-0.031*down1$dist)

down2 = pbp[(pbp$down == 2) & (pbp$playtype %in% c('pass','rush')) & (pbp$yrd2end > 20),]
down2$succ_est = 0.9389*exp(-0.055*down2$dist)

down3 = pbp[(pbp$down == 3) & (pbp$playtype %in% c('pass','rush')) & (pbp$yrd2end > 20),]
down3$succ_est = 0.6256**exp(-0.071*down3$dist)

rbind(down3[,c('gameid','driveid','playid','period','clock','')])